### PR TITLE
op-e2e: action testing L2 sequencer, use L1State in verifier

### PR DESCRIPTION
**Pull request by protolambda**
**Description**

This adds a L2 sequencer (rollup node, wrapping verifier functionality) actor to the action testing in op-e2e.
This actor enables us to test the state-model / logic of sequencing easily, and build a L2 chain with the engine actor for larger action tests later.

This PR depends on the driver refactor in #3647 to pull in the driver state model / logic into the actors. (added PR labels to prevent mergify from combining them accidentally)

**Tests**

To test the sequencer including its L1 origin selection, and cover the code that either forces a repeat of an old origin, or usage of the actual origin selection, I wrote a test for the sequence time drift logic.

**Metadata**

Fix ENG-2762

### CODE REVIEW

File: op-e2e/actions/l2_sequencer.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+	failL2GossipUnsafeBlock error // mock error
+}
+
+func NewL2Sequencer(log log.Logger, l1 derive.L1Fetcher, eng derive.Engine, cfg *rollup.Config, seqConfDepth uint64) *L2Sequencer {
+	ver := NewL2Verifier(log, l1, eng, cfg)
+	return &L2Sequencer{
+		L2Verifier:              *ver,
+		sequencer:               driver.NewSequencer(log, cfg, l1, eng),
+		l1OriginSelector:        driver.NewL1OriginSelector(log, cfg, l1, seqConfDepth),
```
Comment (tynes) [inline op-e2e/actions/l2_sequencer.go:30]: why is the conf depth not on the config? seems like it would be a sane place for it to live

**Review by tynes — APPROVED** [review]

**Comment by trianglesphere** [comment]
@protolambda likely needs rebase

**Review by trianglesphere — COMMENTED** [review]
the conflict should be easily fixed by a rebase + dropping the first two commits

File: op-e2e/actions/l2_sequencer.go (hunk 1 above)
Comment (trianglesphere) [reply op-e2e/actions/l2_sequencer.go:30]: It's the a sequencer specific config. The conf depth is not a property of the rollup, but a property of a sequencer at a point in time (it's valid for the sequencer to change it's conf depth).

File: op-e2e/actions/l2_sequencer.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+		// select origin the real way
+		l1Origin, err := s.l1OriginSelector.FindL1Origin(t.Ctx(), s.l1State.L1Head(), parent)
+		require.NoError(t, err)
+		origin = l1Origin
+	}
+
+	err := s.sequencer.StartBuildingBlock(t.Ctx(), parent, s.derivation.SafeL2Head().ID(), s.derivation.Finalized().ID(), origin)
+	require.NoError(t, err, "failed to start block building")
+
+	s.l2Building = true
```
[the remark this answers is not in the document]
Comment (protolambda) [reply op-e2e/actions/l2_sequencer.go:65]: See `L2Verifier`, it already does.

**Review by trianglesphere — APPROVED** [review]

**Pull request merged**